topics/ui/principles/empty_state.md

```diff
@@ -10,144 +10,261 @@
+
+<table style="none" border="false" column-width="fixed">
+  <tr>
+    <td>
+      <format color="369650" style="bold">Correct</format>
+      <img src="empty_state_clear_reason.png" alt="Correct text in the empty state" />
+    </td>
+    <td >
+      <format color="E55765" style="bold">Incorrect</format>
+      <img src="empty_state_unclear_reason.png" alt="Incorrect text in the empty state" />
+    </td>
+  </tr>
 </table>
 
-Make the reason descriptive:
-
-<format color="Red" style="bold">Incorrect</format>
-
-![](sql-dialect-before.png){width=456 style=block}
-*The word mapping introduces a new entity while there are already two in the table header — path and SQL dialect. In the instructions, it is better to use already existing entities to connect them to what users see on the screen.*
-
-<format color="Green" style="bold">Correct</format>
-
-![](sql-dialect-after.png){width=456 style=block}
-*The word "individual" is used in contrast with the project SQL dialect above the table. The verb specified is used instead of added because, in the table, SQL dialects are not added but selected from a drop-down list in the SQL Dialect column.*
-
-### 2. Actions to fill the area
-
-Required part. An action makes it easier to understand what to start with, instead of searching for the appropriate icon on the toolbar. It can also educate about the shortcut.
-
-Use one or two actions. Three or more actions would make it harder to choose what to start from.
-
-![](maven-tw.png){width=300}
-
-If an action opens a menu, open it at the same position where it would be opened with the corresponding toolbar button. This would explain which toolbar icon opens the menu.
+### Be precise
+
+<p>Do not introduce new entities that are not used in the current context.
+Instead, use the terms from the current UI so that the user understands
+what exactly is missing:</p>
+
+<table style="none" border="false" column-width="fixed">
+  <tr>
+    <td>
+      <format color="369650" style="bold">Correct</format>
+      <img src="empty_state_clear_reason.png" alt="The reason is clear"/>
+    </td>
+    <td>
+      <format color="E55765" style="bold">Incorrect</format>
+      <img src="empty_state_unclear_reason.png" alt="The reason is not clear"/></td>
+  </tr>
+</table>
 
-If an action cannot be tied to a link, explain what to do.
+## Actions to fill the area
 
-![](todo-tw.png){width=574}
+Required part. An action makes it easier to understand what to start with
+instead of searching for the appropriate icon on the toolbar.
+It can also educate about the shortcut.
 
-Hide the area's toolbar if it does not have the same action as in the empty state. Usually, all other toolbar actions are not relevant in this case.
+### Use one or two actions
 
-<format color="Red" style="bold">Incorrect</format>
+Three or more actions would make it harder to choose what to start from.
```

Comment:
We have a couple of cases where we list 3 actions, like in the Run tool window. And I think it's an ok case to be mentioned. Possibly the guideline could be "In rare cases, when you need to list all possible starting points, use the list layout."